client/app/lib/flux/teams/actions.coffee

```diff
@@ -301,6 +301,26 @@ leaveTeam = (partial) ->
             global.location.replace '/'
 
 
+deleteTeam = (partial) ->
+
+  new Promise (resolve, reject) ->
+    new VerifyPasswordModal 'Confirm', partial, (currentPassword) ->
+
+      whoami().fetchEmail (err, email) ->
+        options = { password: currentPassword, email }
+        remote.api.JUser.verifyPassword options, (err, confirmed) ->
```

Comment:
I added that check to backend too
https://github.com/koding/koding/pull/10521/files#diff-23aa32e384f44c80809d4a6cb4c27095R1519

I did this check in front-end because we were doing the same thing for leave team feature as well.